packages/eslint-plugin-turbo/__tests__/lib/env-var-dependencies.test.ts

```diff
@@ -0,0 +1,132 @@
+import rule from "../../lib/rules/no-uncached-env-vars";
+import { RuleTester } from "eslint";
+
+const ruleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 2020 },
+});
+
+const getTestTurboConfig = () => {
+  return {
+    $schema: "./docs/public/schema.json",
+    baseBranch: "origin/main",
+    globalDependencies: ["$GLOBAL_ENV_KEY"],
+    pipeline: {
+      test: {
+        outputs: ["coverage/**/*"],
+        dependsOn: ["^build"],
+      },
+      lint: {
+        outputs: [],
+      },
+      dev: {
+        cache: false,
+      },
+      build: {
+        outputs: ["dist/**/*", ".next/**/*"],
+        dependsOn: ["^build", "$TASK_ENV_KEY", "$ANOTHER_ENV_KEY"],
+      },
+    },
+  };
+};
+
+ruleTester.run("env-var-dependencies", rule, {
+  valid: [
+    {
+      code: `
+        const { TASK_ENV_KEY, ANOTHER_ENV_KEY } = process.env;
+      `,
+      options: [{ turboConfig: getTestTurboConfig() }],
+    },
+    {
+      code: `
+        const x = process.env.GLOBAL_ENV_KEY;
+        const { TASK_ENV_KEY, GLOBAL_ENV_KEY: renamedX } = process.env;
+      `,
+      options: [{ turboConfig: getTestTurboConfig() }],
+    },
+    {
+      code: "var x = process.env.GLOBAL_ENV_KEY;",
+      options: [{ turboConfig: getTestTurboConfig() }],
+    },
+    {
+      code: "let x = process.env.TASK_ENV_KEY;",
+      options: [{ turboConfig: getTestTurboConfig() }],
+    },
+    {
+      code: "const x = process.env.ANOTHER_KEY_VALUE;",
+      options: [
+        {
+          turboConfig: getTestTurboConfig(),
+          allowList: ["^ANOTHER_KEY_[A-Z]+$"],
+        },
+      ],
+    },
+    {
+      code: `
+      var x = process.env.ENV_VAR_ONE;
+      var y = process.env.ENV_VAR_TWO;
+      `,
+      options: [
+        {
+          turboConfig: getTestTurboConfig(),
+          allowList: ["^ENV_VAR_[A-Z]+$"],
+        },
+      ],
+    },
+    {
+      code: `
+      var x = process.env.ENV_VAR_ONE;
+      var y = process.env.ENV_VAR_TWO;
+      `,
+      options: [
+        {
+          turboConfig: getTestTurboConfig(),
+          allowList: ["^ENV_VAR_O[A-Z]+$", "ENV_VAR_TWO"],
+        },
+      ],
+    },
```

Comment:
For bonus points:
- `var fooOrBar = process.env.FOO || process.env.BAR;`
- `() => { return process.env.FOO; }`
- `var foo = process?.env.FOO;`
- `function(arg1 = process.env.FOO) {}`

Optional:
- `(env => env.FOO)(process.env)`
- `var env = process.env; env.FOO;`
- `var process = { env: { TROLL: 'lol' } };`

Reply:
Including bonus points, optional will take a little more logic beyond just AST's and I'm going to punt from the first release 😄

Reply:
Also, just looking at `(env => env.FOO)(process.env)` makes me angry.

Reply:
This whole thing was "how much of a troll can I be?" 😜